src/io/iter_image_recordio_2.cc

```diff
@@ -363,7 +363,7 @@ inline void ImageRecordIOParser2<DType>::ParseChunk(dmlc::InputSplit::Blob * chu
       for (int i = 0; i < res.rows; ++i) {
         uchar* im_data = res.ptr<uchar>(i);
         for (int j = 0; j < res.cols; ++j) {
-          DType RGBA[4];
+          DType* RGBA = new DType[n_channels];
```

Comment:
This seems to be allocating new space for each pixel. Would it be possible to alloc once and reuse the RGBA buffer for all pixels?